typescript/sdk/src/deploy/HyperlaneAppChecker.ts

```diff
@@ -81,41 +88,59 @@ export abstract class HyperlaneAppChecker<
 
   async checkProxiedContracts(
     chain: ChainName,
-    proxyAdminAddress?: Address,
+    owner: Address,
+    ownableOverrides?: Record<string, Address>,
   ): Promise<void> {
-    const expectedProxyAdminAddress =
-      proxyAdminAddress ?? this.app.getContracts(chain).proxyAdmin.address;
-    if (!expectedProxyAdminAddress) {
-      throw new Error(
-        `Checking proxied contracts for ${chain} with no admin provided`,
-      );
-    }
     const provider = this.multiProvider.getProvider(chain);
     const contracts = this.app.getContracts(chain);
 
-    const expectedProxyAdminContract = ProxyAdmin__factory.connect(
-      expectedProxyAdminAddress,
-      provider,
-    );
-
     await promiseObjAll(
       objMap(contracts, async (name, contract) => {
         if (await isProxy(provider, contract.address)) {
-          // Check the ProxiedContract's admin matches expectation
           const actualProxyAdminAddress = await proxyAdmin(
             provider,
             contract.address,
           );
-          if (!eqAddress(actualProxyAdminAddress, expectedProxyAdminAddress)) {
-            this.addViolation({
-              type: ViolationType.ProxyAdmin,
+
+          // get owner of proxy admin contract
+          const actualProxyAdminContract = ProxyAdmin__factory.connect(
+            actualProxyAdminAddress,
+            provider,
+          );
+          const actualProxyAdminOwner = await actualProxyAdminContract.owner();
+          const expectedOwner = ownableOverrides?.['proxyAdmin'] ?? owner;
+
+          if (!eqAddress(actualProxyAdminOwner, expectedOwner)) {
```

Comment:
This will check any proxyAdmin (AW or third party) is owned by the expected owner in the config, we are pulling the proxyAdmin owner check out of `checkOwnership` and doing it here instead.